backend/api_v2/api_deployment_views.py

```diff
@@ -73,21 +74,26 @@ def post(
     def get(
         self, request: Request, org_name: str, api_name: str, api: APIDeployment
     ) -> Response:
-        execution_id = request.query_params.get("execution_id")
-        include_metadata = (
-            request.query_params.get(ApiExecution.INCLUDE_METADATA, "false").lower()
-            == "true"
-        )
-        if not execution_id:
-            raise InvalidAPIRequest("execution_id shouldn't be empty")
-        response: ExecutionResponse = DeploymentHelper.get_execution_status(
-            execution_id=execution_id
-        )
+        serializer = ExecutionQuerySerializer(data=request.query_params)
+        serializer.is_valid(raise_exception=True)
+
+        execution_id = serializer.validated_data.get(ApiExecution.EXECUTION_ID)
+        include_metadata = serializer.validated_data.get(ApiExecution.INCLUDE_METADATA)
+
+        try:
+            response: ExecutionResponse = DeploymentHelper.get_execution_status(
+                execution_id
+            )
+        except InvalidAPIRequest as e:
```

Comment:
@pk-zipstack where does this exception `InvalidAPIRequest` get raised from? I doubt if its being raised from within this code block so this exception handling will not be required. Also, since its a DRF exception already - we can just let it propagate up the stack and get handled by our middleware ultimately